src/envoy/http/mixer/filter.cc

```diff
@@ -155,21 +155,16 @@ void Filter::completeCheck(const CheckResponseInfo& info) {
     return;
   }
 
+  route_directive_ = info.route_directive;
+
+  // set UAEX access log flag
   if (!status.ok()) {
-    state_ = Responded;
-    int status_code = ::istio::utils::StatusHttpCode(status.error_code());
-    decoder_callbacks_->sendLocalReply(Code(status_code), status.ToString(),
-                                       nullptr, absl::nullopt);
     decoder_callbacks_->streamInfo().setResponseFlag(
         StreamInfo::ResponseFlag::UnauthorizedExternalService);
-    return;
   }
 
-  state_ = Complete;
-  route_directive_ = info.route_directive;
-
   // handle direct response from the route directive
-  if (status.ok() && route_directive_.direct_response_code() != 0) {
+  if (route_directive_.direct_response_code() != 0) {
```

Comment:
does this mean we respond with `LocalReply` even when status is not ok?

Reply:
If there is direct response code, yes. Normally, it's not set. If there is an error detail, Mixer will set it for the error case.